Rename int4 to 8da4w in llama2 quantization

andrewor14 opened the pull request:
Summary:
int4 has been confused with "int4 weight only" before, when
in reality it is "int4 weights + int8 dynamic activations".
Renaming it to "8da4w" will reduce confusion and make it
more consistent with "8da4w-gptq".

Differential Revision: D55215146

jerryzh168 reviewed (approved):
thanks @andrewor14

jerryzh168 commented:
I'm also doing a refactor, but please land this one first

jerryzh168 commented on examples/models/llama2/model.py:
```diff
@@ -142,8 +142,8 @@ def __init__(self, **kwargs):
             self.model_ = simple_quantizer.convert_for_runtime()
-        elif "int4" in str(checkpoint_path):
-            print("Using int4 weight-only quantization!")
+        elif "int4" in str(checkpoint_path) or "8da4w" in str(checkpoint_path):
```
we can remove int4?

jerryzh168 commented on examples/models/llama2/export_llama_lib.py:
```diff
@@ -466,7 +466,7 @@ def _prepare_for_llama_export(modelname: str, args) -> LlamaEdgeManager:
         dtype_override = DType[args.dtype_override]
     else:
-        dtype_override = DType["fp16"] if args.quantization_mode == "int4" else None
+        dtype_override = DType["fp16"] if args.quantization_mode == "8da4w" else None
```
probably should add 8da4w-gptq as well

andrewor14 committed:
Rename int4 to 8da4w in llama2 quantization (#2573)

Summary:

int4 has been confused with "int4 weight only" before, when
in reality it is "int4 weights + int8 dynamic activations".
Renaming it to "8da4w" will reduce confusion and make it
more consistent with "8da4w-gptq".

Reviewed By: jerryzh168

Differential Revision: D55215146
